beacon-chain/operations/service_test.go

```diff
@@ -164,6 +166,109 @@ func TestHandleAttestation_Saves_NewAttestation(t *testing.T) {
 	}
 }
 
+func TestHandleAttestation_Aggregates_LargeNumValidators(t *testing.T) {
+	beaconDB := dbutil.SetupDB(t)
+	defer dbutil.TeardownDB(t, beaconDB)
+	ctx := context.Background()
+	broadcaster := &mockBroadcaster{}
+	opsSrv := NewOpsPoolService(ctx, &Config{
+		BeaconDB: beaconDB,
+		P2P:      broadcaster,
+	})
+
+	// First, we create a common attestation data.
+	data := &ethpb.AttestationData{
+		Source: &ethpb.Checkpoint{Epoch: 0, Root: []byte("hello-world")},
+		Target: &ethpb.Checkpoint{Epoch: 0},
+		Crosslink: &ethpb.Crosslink{
+			Shard:      1,
+			StartEpoch: 0,
+		},
+	}
+	dataAndCustodyBit := &pb.AttestationDataAndCustodyBit{
+		Data:       data,
+		CustodyBit: false,
+	}
+	root, err := ssz.HashTreeRoot(dataAndCustodyBit)
+	if err != nil {
+		t.Error(err)
+	}
+	att := &ethpb.Attestation{
+		Data:        data,
+		CustodyBits: bitfield.Bitlist{0x00, 0x00, 0x00, 0x00, 0x01},
+	}
+
+	// We setup the genesis state with 256 validators.
+	deposits, privKeys := testutil.SetupInitialDeposits(t, 256)
+	beaconState, err := state.GenesisBeaconState(deposits, uint64(0), &ethpb.Eth1Data{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	stateRoot, err := ssz.HashTreeRoot(beaconState)
+	if err != nil {
+		t.Fatal(err)
+	}
+	block := blocks.NewGenesisBlock(stateRoot[:])
+	blockRoot, err := ssz.HashTreeRoot(block)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := beaconDB.SaveBlock(ctx, block); err != nil {
+		t.Fatal(err)
+	}
+	if err := beaconDB.SaveHeadBlockRoot(ctx, blockRoot); err != nil {
+		t.Fatal(err)
+	}
+	if err := beaconDB.SaveState(ctx, beaconState, blockRoot); err != nil {
+		t.Fatal(err)
+	}
+
+	// Next up, we compute the committee for the attestation we're testing.
+	committee, err := helpers.CrosslinkCommittee(beaconState, att.Data.Target.Epoch, att.Data.Crosslink.Shard)
+	if err != nil {
+		t.Error(err)
+	}
+	attDataRoot, err := ssz.HashTreeRoot(att.Data)
+	if err != nil {
+		t.Error(err)
+	}
+	totalAggBits := bitfield.NewBitlist(uint64(len(committee)))
+
+	// For every single member of the committee, we sign the attestation data and handle
```

Comment:
for the test can we try to emulate how it will be in production a bit more ? We will have like 256 attestations at the same time, can we make it concurrent instead of single threaded ? so the signing , and sending of it to ops service can be done in a separate goroutine for each validator